lib/DXIL/DxilUtilDbgInfoAndMisc.cpp

```diff
@@ -75,7 +86,7 @@ Value *MergeGEP(GEPOperator *SrcGEP, GEPOperator *GEP) {
     }
 
     // Update the GEP in place if possible.
-    if (SrcGEP->getNumOperands() == 2) {
+    if (SrcGEP->getNumOperands() == 2 && !AsCast) {
```

Comment:
Do we ever hit this case where `Sum` is non-zero, especially if `GO1` was non-zero?

Reply:
I think I hit it in one of the test cases I was writing at some point. But its not clear we can reach that case from hlsl. Probably we can, but it has yet to be seen in practice.

Reply:
Normally, we don't use the first index for anything in HLSL - it's always zero.  However, IIRC, I saw these used once when generated for array element construction or something like that.

Reply:
I think it is not too hard to break this code by writing some custom IR test cases, but I'm not sure its worth hardening this code against unexpected inputs.